### PR TITLE
Use --env=super by default for Linuxbrew

### PR DESCRIPTION
**Pull request by sjackman**

### CODE REVIEW

File: Library/Homebrew/extend/ENV.rb [hunk 1]
```diff
@@ -4,11 +4,7 @@
 require "extend/ENV/super"
 
 def superenv?
-  if OS.linux?
-    ARGV.env == "super" || ARGV.homebrew_developer? && ARGV.env != "std"
```
Comment (maxim-belkin) [inline Library/Homebrew/extend/ENV.rb]: Wasn't this line ambiguous?
`1 || (2 && 3)` and `(1 || 2) && 3`

**Review by maxim-belkin — APPROVED** [review]

File: Library/Homebrew/extend/ENV.rb (hunk 1 above)
Comment (sjackman) [reply Library/Homebrew/extend/ENV.rb]: `&&` has higher precedence than `||`. http://phrogz.net/programmingruby/language.html#table_18.4

Comment (maxim-belkin) [reply Library/Homebrew/extend/ENV.rb]: What was the purpose of `&& ARGV.env != "std"`?

Comment (sjackman) [reply Library/Homebrew/extend/ENV.rb]: When `--env=std` is specified use `ENV/std` and ignore the value of `ARGV.homebrew_developer?`

**Pull request merged**